### PR TITLE
Save screenshots and gifs of the current canvas 

### PR DESCRIPTION
**Pull request by claban3**

### CODE REVIEW

File: src/Components/Canvas/Canvas.tsx [hunk 1]
```diff
@@ -1,98 +1,176 @@
...
+            if (record == true) {
+                record = false;
+                if (gif) gif.render();
+                frameCounter = 0;
+            }
+
+            record = props.canvasSettings.record;
+            setupGif(setRecordCanvasInParent);
+        }
+
```
Comment (cannono) [inline src/Components/Canvas/Canvas.tsx:173]: I don't think this is working as expected. If you stop the recording early it continues to add frames while rendering the gif.

**Review by cannono — CHANGES_REQUESTED** [review]

File: src/Components/Canvas/Canvas.tsx (hunk 1 above)
Comment (claban3) [reply src/Components/Canvas/Canvas.tsx:173]: can you try again, this is definitely working correctly for me?

Comment (cannono) [reply src/Components/Canvas/Canvas.tsx:173]: Add a console.log() on line 115. It still prints while the gif is rendering.

Comment (claban3) [reply src/Components/Canvas/Canvas.tsx:173]: this has been fixed. However, we really shouldn't be using javascript numbers to store state any more please.

**Review by cannono — APPROVED** [review]

**Review by aposp — APPROVED** [review]
Looks good to me. @claban3 can you resolve the merge conflict and merge to main? I'll keep testing to make sure I understand how to best demo it tomorrow.

**Pull request merged**